api/applicationserver_integrations_storage.proto

```diff
@@ -82,6 +80,43 @@ message GetStoredApplicationUpRequest {
   google.protobuf.Duration last = 10;
 }
 
+message GetStoredApplicationUpCountRequest {
+  // Count upstream messages from all end devices of an application. Cannot be used in conjunction with end_device_ids.
+  ApplicationIdentifiers application_ids = 1;
+  // Count upstream messages from a single end device. Cannot be used in conjunction with application_ids.
+  EndDeviceIdentifiers end_device_ids = 2;
+
+  // Count upstream messages of a specific type. If not set, then all upstream messages are returned.
+  string type = 3 [(validate.rules).string = { in: [
+    "",
+    "uplink_message",
+    "join_accept",
+    "downlink_ack",
+    "downlink_nack",
+    "downlink_sent",
+    "downlink_failed",
+    "downlink_queued",
+    "downlink_queue_invalidated",
+    "location_solved",
+    "service_data"
+  ] }];
+
+  // Count upstream messages after this timestamp only. Cannot be used in conjunction with last.
+  google.protobuf.Timestamp after = 4;
+  // Count upstream messages before this timestamp only. Cannot be used in conjunction with last.
+  google.protobuf.Timestamp before = 5;
+  // Count uplinks on a specific FPort only.
+  google.protobuf.UInt32Value f_port = 6;
+
+  // Count upstream messages that have arrived in the last minutes or hours. Cannot be used in conjunction with after and before.
+  google.protobuf.Duration last = 7;
+}
+
+
+message GetStoredApplicationUpCountResponse {
+  map<string,uint32> count = 1;
```

Comment:
The map key here is the type, right? Please document that here

Reply:
This map key is the device ID, so the output looks like:

```json
{"count": {"dev1": 100, "dev2": 200}}
```

All requests are scoped for a specific application and tenant.